google/cloud/testing_util/validate_metadata.cc

```diff
@@ -321,7 +320,11 @@ void ValidateMetadataFixture::IsContextMDValid(
 
   auto const* method =
       DescriptorPool::generated_pool()->FindMethodByName(method_name);
-  ASSERT_THAT(method, NotNull()) << "Method " + method_name + " is unknown.";
+  if (method == nullptr) {
+    GCP_LOG(INFO) << "`x-goog-request-params` header not verified for "
```

Comment:
When running this from the command-line, logging is likely disabled.  I wonder if there is a way to log from googletest.  `SUCCEED() << "msg"` ?  Or `SCOPED_TRACE`?

Reply:
Should we worry that misspelled method names will not get caught?

We can rewrite the IAM credentials stub test to use `ValidateMetadataFixture::GetMetadata` instead of `ValidateMetadataFixture::IsContextMDValid` to not lose any test coverage. (In fact, we could gain some).

Reply:
> Should we worry that misspelled method names will not get caught?

Perhaps?  But the response may well be to obtain the name via reflection rather than typing it out.

> We can rewrite the IAM credentials stub test to use `ValidateMetadataFixture::GetMetadata` instead of `ValidateMetadataFixture::IsContextMDValid` to not lose any test coverage. (In fact, we could gain some).

In this PR we're strictly gaining some test coverage, so I'll leave any question of how/what we should be testing to some future PR.